src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1152,6 +1152,12 @@ private bool isValidStringRef(CORINFO_MODULE_STRUCT_* module, uint metaTOK)
         private bool shouldEnforceCallvirtRestriction(CORINFO_MODULE_STRUCT_* scope)
         { throw new NotImplementedException("shouldEnforceCallvirtRestriction"); }
 
+        private byte* getStringLiteral(CORINFO_MODULE_STRUCT_* module, uint metaTOK, ref int length)
+        {
+            length = -1;
```

Comment:
This can still return valid length here without too much hassle.

Reply:
This should be the full implementation:

```csharp
        private byte* getStringLiteral(CORINFO_MODULE_STRUCT_* module, uint metaTOK, ref int length)
        {
            MethodIL methodIL = (MethodIL)HandleToObject((IntPtr)module);
            string s = (string)methodIL.GetObject(metaTOK);
            length = s.Length;
            return (byte*)GetPin(StringToUTF16(s));
        }

        private static byte[] StringToUTF16(string s)
        {
            int byteCount = Encoding.UTF16.GetByteCount(s);
            byte[] bytes = new byte[byteCount + 2];
            Encoding.UTF16.GetBytes(s, 0, s.Length, bytes, 0);
            return bytes;
        }
```

Reply:
> return (byte*)GetPin(StringToUTF16(s));

This should be just `return (byte*)GetPin(s);`.

Reply:
thanks!